<div class="container">

<table style="width: 100%;"><tr>
<td>haps_ssites</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Organize information to create haplotypes using segregating sites matrices</h2>

<h3>Description</h3>

<p>This function organizes higher-level information for creating haplotypes from
matrices of segregating sites output from coalescent simulations.
</p>


<h3>Usage</h3>

<pre><code class="language-R">haps_ssites(obj = NULL, fn = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>Object containing segregating sites information.
This can be one of the following:
(1) A single <code>list</code> with a <code>seg_sites</code> field inside. This field must
contain a matrix for segregating sites for each chromosome.
The matrix itself should contain the haplotype information, coded
using 0s and 1s: 0s indicate the ancestral state and 1s indicate
mutant.
The matrix column names should indicate the positions of the polymorphisms on the
chromosome.
If positions are in the range <code style="white-space: pre;">⁠(0,1)⁠</code>, they're assumed to come from an infinite-
sites model and are relative positions.
If positions are integers in the range <code style="white-space: pre;">⁠[0, chromosome length - 1]⁠</code>
or <code style="white-space: pre;">⁠[1, chromosome length]⁠</code>, they're assumed to come from an finite-sites
model and are absolute positions.
Defaults to <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fn</code></td>
<td>
<p>A single string specifying the name of the file containing
the <code>ms</code>-style coalescent output with segregating site info.
Defaults to <code>NULL</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For what the <code>seg_sites</code> field should look like in a list, see output from the
<code>scrm</code> or <code>coala</code> package.
(These packages are not required to be installed when installing
<code>jackalope</code>.)
If using either of these packages, I encourage you to cite them. For citation
information, see output from <code>citation("scrm")</code> or <code>citation("coala")</code>.
</p>


<h3>Value</h3>

<p>A <code>haps_ssites_info</code> object containing information used in <code>create_haplotypes</code>
to create variant haplotypes.
This class is just a wrapper around a list of matrices of segregating site info,
which you can view (but not change) using the object's <code>mats()</code> method.
</p>


</div>