<div class="container">

<table style="width: 100%;"><tr>
<td>create_genome</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create a reference genome.</h2>

<h3>Description</h3>

<p>Random chromosomes are generated to create a new <code>ref_genome</code> object.
Note that this function will never generate empty chromosomes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">create_genome(
  n_chroms,
  len_mean,
  len_sd = 0,
  pi_tcag = rep(0.25, 4),
  n_threads = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n_chroms</code></td>
<td>
<p>Number of chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>len_mean</code></td>
<td>
<p>Mean for the gamma distribution of chromosome sizes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>len_sd</code></td>
<td>
<p>Standard deviation for the gamma distribution of chromosome sizes.
If set to <code>&lt;= 0</code>, all chromosomes will be the same length. Defaults to <code>0</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pi_tcag</code></td>
<td>
<p>Vector of length 4 containing the nucleotide equilibrium frequencies
for "T", "C", "A", and "G", respectively. Defaults to <code>rep(0.25, 4)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_threads</code></td>
<td>
<p>Number of threads to use for parallel processing. This argument is
ignored if OpenMP is not enabled. Defaults to <code>1</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code>ref_genome</code> object.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
genome &lt;- create_genome(10, 100e3, 100, pi_tcag = c(0.1, 0.2, 0.3, 0.4))

</code></pre>


</div>