<div class="container">

<table style="width: 100%;"><tr>
<td>haps_phylo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Organize information to create haplotypes using phylogenetic tree(s)</h2>

<h3>Description</h3>

<p>This function organizes higher-level information for creating haplotypes from
phylogenetic tree(s) output as <code>phylo</code> or <code>multiPhylo</code> objects
(both from the <code>ape</code> package) or NEWICK files.
Note that all phylogenetic trees must be rooted and binary.
If using this function, I encourage you to cite <code>ape</code>. For citation
information, see output from <code>citation("ape")</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">haps_phylo(obj = NULL, fn = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>Object containing phylogenetic tree(s).
This can be (1) a single <code>phylo</code> object
that represents all chromosomes in the genome or
(2) a <code>list</code> or <code>multiPhylo</code> object containing a <code>phylo</code> object for
each reference chromosome.
In the latter case, phylogenies will be assigned to chromosomes in the
order provided.
Defaults to <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fn</code></td>
<td>
<p>One or more string(s), each of which specifies the file name
of a NEWICK file containing a phylogeny.
If one name is provided, that phylogeny will be used for all chromosomes.
If more than one is provided, there must be a phylogeny for each reference
genome chromosome, and phylogenies will be assigned to chromosomes
in the order provided.
Defaults to <code>NULL</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>See <code>?ape::write.tree</code> for writing phylogenies to an output file.
</p>


<h3>Value</h3>

<p>A <code>haps_phylo_info</code> object containing information used in <code>create_haplotypes</code>
to create variant haplotypes.
This class is just a wrapper around a list containing phylogenetic tree
information for each reference chromosome, which you can view (but not change)
using the object's <code>phylo()</code> method.
</p>


</div>