<div class="container">

<table style="width: 100%;"><tr>
<td>SimulatedPedigree</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Simulated SNP data according to a Pedigree
</h2>

<h3>Description</h3>

<p>A matrix containing 111 individuals (rows), pedigree information (first five columns) and 20.000 single
nucleotide polymorphisms (remaining columns) coded in (0,1,2) format and simulated according to a
pedigree of 20 unrelated founders with six posterior generations. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("SimulatedPedigree")</code></pre>


<h3>Format</h3>

<p>A data frame containing 111 rows and 20.005 columns.</p>


<h3>Details</h3>

<p>The SNP data was generated using the JGTeach package. The genotype counts in the object SimulatedPedigree
represent the counts of the minor allele.
</p>


<h3>References</h3>

<p>Goudet, J. (2022) JGTeach: JG Teaching material. R package version 0.1.9. https://github.com/jgx65
</p>
<p>Graffelman, J., Weir, B.S. and Goudet, J. (2024) Estimation of Jacquard's genetic identity coefficients with
bi-allelic variants by constrained least-squares. Preprint at bioRxiv <a href="https://doi.org/10.1101/2024.03.25.586682">doi:10.1101/2024.03.25.586682</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(SimulatedPedigree)
</code></pre>


</div>