<div class="container">

<table style="width: 100%;"><tr>
<td>Jacquard.cls</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Estimation of Jacquard coefficients by Constrained Least Squares
</h2>

<h3>Description</h3>

<p>Function <code>Jacquard.cls</code> estimates the nine condensed Jacquard coefficients of a pair of individuals
using their joint genotype probabilities and the allele frequencies using constrained least squares.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Jacquard.cls(Xlist, mafvec = NULL, eps = 1e-06, delta.init = runif(9), Mavg = NULL,
             inner.iter = 1000, outer.iter = 1000, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Xlist</code></td>
<td>
<p>A list object with nine fields containing the matrices with joint genotype counts.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mafvec</code></td>
<td>
<p>A vector with the minor allele frequencies for all genetic variants.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eps</code></td>
<td>
<p>Tolerance criterion for the solver (<code>eps=1e-06</code> by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delta.init</code></td>
<td>
<p>Initial vector of estimates for the nine condensed Jacquard coefficients.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Mavg</code></td>
<td>
<p>A nine by nine matrix of conditional probabilities, allele frequency dependent. This matrix
is calculated by <code>Jacquard.cls</code> if it is not specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inner.iter</code></td>
<td>
<p>Maximum number of inner iterations for the solver (1000 by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outer.iter</code></td>
<td>
<p>Maximum number of outer iterations for the solver (1000 by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Print output on the progress of the algorithm if <code>verbose=TRUE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Function <code>Jacquard.cls</code> relies on the solver <code>solnp</code> from the Rsolnp package.
</p>


<h3>Value</h3>

<p>A list object with fields:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>delta</code></td>
<td>
<p>A list with nine matrices of estimates of pairwise Jacquard coefficients.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>convergence</code></td>
<td>
<p>A matrix with the convergence status for each pair (0 = converged; 1 = not converged).</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>References</h3>

<p>Graffelman, J., Weir, B.S. and Goudet, J. (2024) Estimation of Jacquard's genetic identity coefficients with
bi-allelic variants by constrained least-squares. Preprint at bioRxiv <a href="https://doi.org/10.1101/2024.03.25.586682">doi:10.1101/2024.03.25.586682</a>.
</p>
<p>Ghalanos, A. and Theussl, S. (2015) Rsolnp: General Non-linear Optimization Using Augmented Lagrange Multiplier Method.
R package version 1.16. <a href="https://cran.r-project.org/package=Rsolnp">https://cran.r-project.org/package=Rsolnp</a>
</p>


<h3>See Also</h3>

<p><code>JointGenotypeCounts</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(SimulatedPedigree)
Xgen &lt;- as.matrix(SimulatedPedigree[,6:ncol(SimulatedPedigree)])

data(GTC)

mafvec &lt;- mafvector(Xgen)

ii &lt;- 1:3

GTCsubset &lt;- list(length = 9)
for (k in 1:9) {
  GTCsubset[[k]] &lt;- matrix(numeric(3^2), ncol = 3)
  GTCsubset[[k]] &lt;- GTC[[k]][ii,ii]
}

output &lt;- Jacquard.cls(GTCsubset,mafvec=mafvec,
                       eps=1e-06)
Delta.cls &lt;- output$delta
print(Delta.cls)
print(output$convergence)

# A particular estimate of a Jacquard coefficient for a particular pair can
# be extracted from Delta.cls
#
# E.g., Delta\_9 of the first pair of individuals (1,2) can be extracted by
#

D9_12 &lt;- Delta.cls[[9]][1,2]

</code></pre>


</div>