<div class="container">

<table style="width: 100%;"><tr>
<td>track_alignments</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create an AlignmentsTrack for a custom JBrowse 2 view</h2>

<h3>Description</h3>

<p>Creates the necessary configuration string for an
indexed BAM or CRAM alignment so that it can be used
in a JBrowse custom linear genome view.
</p>


<h3>Usage</h3>

<pre><code class="language-R">track_alignments(track_data, assembly)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>track_data</code></td>
<td>
<p>the URL to the BAM/CRAM alignments</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assembly</code></td>
<td>
<p>the config string generated by <code>assembly</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>It is important to note that while only the BAM/CRAM file is
passed as an argument, <code>tracks_alignment</code> assumes that a BAM/CRAM
index of the same name is located with the file
</p>
<p>For example:
</p>
<p><code>track_alignments("data/alignments.bam")</code>
</p>
<p>Assumes that <code>data/alignments.bam.bai</code> also exists.
</p>
<p>This is a JBrowse 2 convention, and the default naming output of samtools
</p>
<p>For more information on creating an index with samtools, visit
<a href="https://www.htslib.org/">https://www.htslib.org/</a>
</p>


<h3>Value</h3>

<p>a character vector of stringified AlignmentsTrack JSON configuration
</p>


<h3>Examples</h3>

<pre><code class="language-R">assembly &lt;- assembly("https://jbrowse.org/genomes/hg19/fasta/hg19.fa.gz", bgzip = TRUE)

track_alignments("alignments.bam", assembly)
track_alignments("alignments.cram", assembly)
</code></pre>


</div>