<div class="container">

<table style="width: 100%;"><tr>
<td>plotSeg</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot signal and breakpoints with segment-level signal estimates</h2>

<h3>Description</h3>

<p>Plot signal and breakpoints with segment-level signal estimates
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotSeg(dat, breakpoints = NULL, regNames = NULL,
  exclNames = c("genotype", "region", "bT", "bN", "cellularity"),
  ylabs = colnames(dat), ylims = NULL, binExclPattern = "^b[N|T]*$",
  col = "#33333333", pch = 19, cex = 0.3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dat</code></td>
<td>
<p>A <code>matrix</code> or data frame whose rows correspond to loci
sorted along the genome, or a <code>numeric</code> <code>vector</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>breakpoints</code></td>
<td>
<p>A vector of breakpoints positions, or a <code>list</code> of
such vectors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>regNames</code></td>
<td>
<p>Region labels, a vector of length
<code>length(breakpoints)+1</code> (if <code>breakpoints</code> is a vector) or of
length <code>length(breakpoints[[1]])+1</code> (if <code>breakpoints</code> is a list).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclNames</code></td>
<td>
<p>A vector of column names corresponding to columns that
should not be plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylabs</code></td>
<td>
<p>A vector of 'y' labels (column names or indices) that should be
plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylims</code></td>
<td>
<p>An optional <code class="reqn">2*d</code> matrix with <code>ylim</code> values for each
of the <code class="reqn">d</code> dimensions to be plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>binExclPattern</code></td>
<td>
<p>A vector of column names or indices in
<code>colnames(dat)</code> for which segment-level signal estimates should *not*
be drawn.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>Color of plotting symbol, see <code>par</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pch</code></td>
<td>
<p>Plotting symbol, see <code>par</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>Magnification factor for plotting symbol, see <code>par</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Argument 'binCols' is mainly used to avoid calculating mean levels for
allelic ratios, which would not make sense as they are typically
multimodal.
</p>


<h3>Author(s)</h3>

<p>Morgane Pierre-Jean and Pierre Neuvial
</p>


<h3>Examples</h3>

<pre><code class="language-R">
affyDat &lt;- acnr::loadCnRegionData(dataSet="GSE29172", tumorFraction=1)
sim &lt;- getCopyNumberDataByResampling(1e4, 5, minLength=100, regData=affyDat)
dat &lt;- sim$profile
res &lt;- PSSeg(dat, method="RBS", stat=c("c", "d"), K=50)
bkpList &lt;- list(true=sim$bkp, est=res$bestSeg)
plotSeg(dat, breakpoints=bkpList)
</code></pre>


</div>